### DISCUSSION TITLE
minimum geosparql example

### DISCUSSION THREAD

**Discussion by luisramosbosch** [eclipse-rdf4j/rdf4j#4504] [Q&A] [answered]
Dear All, 

I am trying to develop a minimum example in geosparql, following this documentation:

https://rdf4j.org/documentation/programming/geosparql/

I have the idea, that I should be able to build the example using code, but when I tried to do it, 
I found some method (object property, and class defition) I would expect were no present,
like geo:hasGeometry or geof:Point, which would be neccesary to build the model of the example.

model.add(eiffelTower, GEO.hasGeometry, coordinateset );
model.add(coordinateset, RDF.TYPE, GEOF.Point );

Could you tell me if building the model from code is possible?, or if I have to edit the model somewhere else, 
and use rdf4j only for querying?.

Best regards

Luis Ramos

**Comment by abrokenjester** [comment, marked as answer]
I am not immediately sure why we don't have those as constants in our vocabulary classes - seems an odd oversight. I've logged a feature request: #4505.

In the meantime though, it shouldn't stop you from creating these models using RDF4J. You can create all relevant IRIs yourself, for example like so:

```java
import static org.eclipse.rdf4j.model.util.Values.iri;

model.add(eiffelTower, iri(GEO.NAMESPACE, "hasGeometry"), coordinateset);
model.add(coordinateset, RDF.TYPE, iri(GEOF.NAMESPACE, "Point"));
```

Doing them inline here for brevity but of course if you plan reusing them, you're best off creating your own reusable constants.

**Comment by luisramosbosch** [comment]
Thanks so much @abrokenjester , I edited the ontology on protégé, uploaded it, and did my basic test.
